clients/src/test/java/org/apache/kafka/common/metrics/stats/RateTest.java

```diff
@@ -64,4 +69,31 @@ public void testRateWithNoPriorAvailableSamples(int numSample, int sampleWindowS
         double expectedRatePerSec = sampleValue / windowSize;
         assertEquals(expectedRatePerSec, observedRate, EPS);
     }
+
+    // Record an event every 100 ms on average, moving some 1 ms back or forth for fine-grained 
+    // window control. The expected rate, hence, is 10-11 events/sec depending on the moment of 
+    // measurement. Start assertions from the second window. This test is to address past issue,
+    // when measurements in the end of the sample led to value spikes.
```

Comment:
How about changing "This test is to address past issue, when measurements in the end of the sample led to value spikes." to sth like "This test covers the case where a sample window partially overlaps with the monitored window." ?

Reply:
Applied